Linking fails with:

```
  = note: /home/servo/.cargo/git/checkouts/gecko-media-6bedb392a81eb7cf/0299b45/src/top.rs:302: error: undefined reference to 'GeckoMedia_Initialize'
          /home/servo/.cargo/git/checkouts/gecko-media-6bedb392a81eb7cf/0299b45/src/top.rs:310: error: undefined reference to 'GeckoMedia_Shutdown'
          /home/servo/.cargo/git/checkouts/gecko-media-6bedb392a81eb7cf/0299b45/src/top.rs:323: error: undefined reference to 'GeckoMedia_ProcessEvents'
          /home/servo/.cargo/git/checkouts/gecko-media-6bedb392a81eb7cf/0299b45/src/top.rs:316: error: undefined reference to 'GeckoMedia_CanPlayType'
```

I have no idea why.

--- a/components/constellation/Cargo.toml
+++ b/components/constellation/Cargo.toml
@@ -43,3 +43,6 @@ webrender_api = {git = "https://github.com/servo/webrender", features = ["ipc"]}
 
 [target.'cfg(all(not(target_os = "windows"), not(target_os = "ios")))'.dependencies]
 gaol = {git = "https://github.com/servo/gaol"}
+
+[target.'cfg(all(any(target_os = "macos", target_os = "linux"), not(any(target_arch = "arm", target_arch = "aarch64"))))'.dependencies]
+gecko-media = {git = "https://github.com/servo/gecko-media.git"}

--- a/components/constellation/constellation.rs
+++ b/components/constellation/constellation.rs
@@ -108,6 +108,8 @@ use debugger;
 use devtools_traits::{ChromeToDevtoolsControlMsg, DevtoolsControlMsg};
 use euclid::{Size2D, TypedSize2D, ScaleFactor};
 use event_loop::EventLoop;
+#[cfg(all(any(target_os = "macos", target_os = "linux"), not(any(target_arch = "arm", target_arch = "aarch64"))))]
+use gecko_media::GeckoMedia;
 use gfx::font_cache_thread::FontCacheThread;
 use gfx_traits::Epoch;
 use ipc_channel::{Error as IpcError};
@@ -1481,6 +1483,16 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
         debug!("Asking compositor to complete shutdown.");
         self.compositor_proxy.send(ToCompositorMsg::ShutdownComplete);
+
+        #[cfg(all(
+            any(target_os = "macos", target_os = "linux"),
+            not(any(target_arch = "arm", target_arch = "aarch64")),
+        ))]
+        {
+            if let Err(()) = GeckoMedia::shutdown() {
+                warn!("Media stack shutdown failed.");
+            }
+        }
     }
 
     fn handle_pipeline_exited(&mut self, pipeline_id: PipelineId) {

--- a/components/constellation/lib.rs
+++ b/components/constellation/lib.rs
@@ -17,6 +17,8 @@ extern crate devtools_traits;
 extern crate euclid;
 #[cfg(all(not(target_os = "windows"), not(target_os = "ios")))]
 extern crate gaol;
+#[cfg(all(any(target_os = "macos", target_os = "linux"), not(any(target_arch = "arm", target_arch = "aarch64"))))]
+extern crate gecko_media;
 extern crate gfx;
 extern crate gfx_traits;
 extern crate hyper;

